File: packages/shared/src/types/application.ts
import { ApplicationStatus } from "./enums";

export interface ApplicationDto {
  id: string;
  jobId: string;
  jobTitle: string;
  /** Başvuran aday mı yoksa taşeron firma mı. */
  applicantType: "CANDIDATE" | "SUBCONTRACTOR";
  /** applicantType'a göre candidateId ya da subcontractorId. */
  applicantId: string;
  /** applicantType'a göre aday adı ya da firma adı. */
  applicantName: string;
  status: ApplicationStatus;
  /** Adayın başvuru sırasında belirttiği beklenen ücret (aylık, TL). */
  expectedWage?: number | null;
  /** Firmanın bu başvuruya özel teklif ettiği ücret (aylık, TL). */
  offeredWage?: number | null;
  createdAt: string;
}

export interface CreateApplicationInput {
  jobId: string;
  message?: string;
  expectedWage?: number;
}

export interface UpdateApplicationStatusInput {
  status: ApplicationStatus;
  offeredWage?: number;
}

File: apps/mobile/lib/models/application.dart
class JobApplication {
  final String id;
  final String jobId;
  final String jobTitle;
  // Başvuran aday mı (CANDIDATE) yoksa taşeron firma mı (SUBCONTRACTOR) — bkz.
  // apps/api ApplicationDto. API eskiden sadece adayları destekliyordu ve bu
  // alanlar candidateId/candidateName idi; taşeron firma başvuruları eklenince
  // applicantType/applicantId/applicantName olarak genelleştirildi.
  final String applicantType;
  final String applicantId;
  final String applicantName;
  final String status; // PENDING | REVIEWED | ACCEPTED | REJECTED
  final String createdAt;

  const JobApplication({
    required this.id,
    required this.jobId,
    required this.jobTitle,
    required this.applicantType,
    required this.applicantId,
    required this.applicantName,
    required this.status,
    required this.createdAt,
  });

  factory JobApplication.fromJson(Map<String, dynamic> json) => JobApplication(
        id: json['id'] as String,
        jobId: json['jobId'] as String,
        jobTitle: json['jobTitle'] as String,
        applicantType: json['applicantType'] as String,
        applicantId: json['applicantId'] as String,
        applicantName: json['applicantName'] as String,
        status: json['status'] as String,
        createdAt: json['createdAt'] as String,
      );
}

const Map<String, String> applicationStatusLabels = {
  'PENDING': 'Beklemede',
  'REVIEWED': 'İncelendi',
  'ACCEPTED': 'Kabul Edildi',
  'REJECTED': 'Reddedildi',
};